### PR TITLE
Pv2 56 payment funding source handle acceptance test

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Steps/FundingSourceStepBase.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Steps/FundingSourceStepBase.cs]: a few unused usings can be removed

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature [hunk 2]
```diff
@@ -3,10 +3,11 @@
 Background:
 
 	Given the current processing period is 10
+	And the payments are for the current collection year
 
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 
-	And the SFA contribution percentage is "90%"
+	And the SFA contribution percentage is 90%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature]: I think that this has been made available in the AcceptanceTests.Core project or is at least the same as the one used in RequiredPayments so could be refactored.

Remove the trailing "

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature [hunk 3]
```diff
@@ -24,7 +25,7 @@ Scenario: Contract Type 2 Learning payment
 
 	When required payments event is received
 
-	Then the payment source component will generate the following contract type 2 Learning (TT1) coinvested payments:
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature]: To match what has been added to RequiredPayments, add back in Learning (TT1) and perhaps rationalise between Funding Source and Required Payments projects

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature [hunk 4]
```diff
@@ -24,7 +25,7 @@ Scenario: Contract Type 2 Learning payment
...
+	Then the payment source component will generate the following contract type 2 coinvested payments:
 
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period  | ULN   | TransactionType | FundingSource			| Amount |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature]: Functionality has been added into master that removes the need to have LearnRefNumber, Ukprn and Uln specified in the table. Check and rationalise between the two projects

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature [hunk 5]
```diff
@@ -3,10 +3,11 @@
 Background:
 
 	Given the current processing period is 10
+	And the payments are for the current collection year
 
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/ALL TT1 TT2 and TT3.feature]: In Required Payments, the whole of the learnref1 value has been made into a step variable, not just the numeric.

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Completion (TT2) and Balancing (TT3).feature [hunk 6]
```diff
@@ -3,10 +3,10 @@
 Background:
 
 	Given the current processing period is 10
-
+	And the payments are for the current collection year
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 
-	And the SFA contribution percentage is "90%"
+	And the SFA contribution percentage is 90%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Completion (TT2) and Balancing (TT3).feature]: remove trailing quotes

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Completion (TT2) and Balancing (TT3).feature [hunk 7]
```diff
@@ -3,10 +3,10 @@
...
 
 	And the required payments component generates the following contract type 2 payable earnings:
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period | ULN   | TransactionType | Amount |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Completion (TT2) and Balancing (TT3).feature]: can remove learnrefnumber, ukprn and uln

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Completion (TT2) only.feature [hunk 8]
```diff
@@ -2,10 +2,10 @@
 
 Background:
 	Given the current processing period is 13
-
+	And the payments are for the current collection year
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 	
-	And the SFA contribution percentage is "90%"
+	And the SFA contribution percentage is 90%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Completion (TT2) only.feature]: remove trailing "

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - ALL TT1 TT2 and TT3.feature [hunk 9]
```diff
@@ -24,7 +24,7 @@ Scenario: Contract Type 2 Learning payment
 
 	When required payments event is received
 
-	Then the payment source component will generate the following contract type 2 Learning (TT1) coinvested payments:
+	Then the payment source component will generate the following contract type 2 coinvested payments:
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - ALL TT1 TT2 and TT3.feature]: Add back in Learning (TT1) to match required payments

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - ALL TT1 TT2 and TT3.feature [hunk 10]
```diff
@@ -3,10 +3,10 @@
 Background:
 
 	Given the current processing period is 10
-
+	And the payments are for the current collection year
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 
-	And the SFA contribution percentage is "100%"
+	And the SFA contribution percentage is 100%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - ALL TT1 TT2 and TT3.feature]: remove trailing "

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - ALL TT1 TT2 and TT3.feature [hunk 11]
```diff
@@ -3,10 +3,10 @@
...
 
 	And the required payments component generates the following contract type 2 payable earnings:
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period | ULN   | TransactionType | Amount |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - ALL TT1 TT2 and TT3.feature]: can remove learnrefnumber, ukprn and uln

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Completion (TT2) only.feature [hunk 12]
```diff
@@ -2,10 +2,10 @@
 
 Background:
 	Given the current processing period is 13
-
+	And the payments are for the current collection year
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 
-	And the SFA contribution percentage is "100%"
+	And the SFA contribution percentage is 100%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Completion (TT2) only.feature]: remove trailing "

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature [hunk 13]
```diff
@@ -2,10 +2,10 @@
 
 Background:
 	Given the current processing period is 2
-
+	And the payments are for the current collection year
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 
-	And the SFA contribution percentage is "100%"
+	And the SFA contribution percentage is 100%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature]: remove trailing "

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature [hunk 14]
```diff
@@ -2,10 +2,10 @@
...
 
 	And the required payments component generates the following contract type 2 payable earnings:
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period | ULN   | TransactionType | Amount |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature]: can remove learnrefnumber, ukprn and uln

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature [hunk 15]
```diff
@@ -19,7 +19,7 @@ Scenario: Contract Type 2 Learning payment
 
 	When required payments event is received
 
-	Then the payment source component will generate the following contract type 2 Learning (TT1) coinvested payments:
+	Then the payment source component will generate the following contract type 2 coinvested payments:
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature]: add back in Learning (TT1) to match required payments

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature [hunk 16]
```diff
@@ -19,7 +19,7 @@ Scenario: Contract Type 2 Learning payment
...
 
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period | ULN   | TransactionType | FundingSource			| Amount |
 	| learnref1      | 10000 | p1                     | 2      | 10000 | Learning (TT1)  | CoInvestedSfa (FS2)		| 600    |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Fully funded - Learning (TT1).feature]: can remove learnrefnumber, ukprn and uln

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - multiple periods.feature [hunk 17]
```diff
@@ -2,10 +2,10 @@
 
 Background:
 	Given the current processing period is 2
-
+	And the payments are for the current collection year
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 
-	And the SFA contribution percentage is "90%"
+	And the SFA contribution percentage is 90%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - multiple periods.feature]: remove trailing "

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - multiple periods.feature [hunk 18]
```diff
@@ -21,7 +21,7 @@ Scenario: Contract Type 2 Learning payment
 
 	When required payments event is received
 
-	Then the payment source component will generate the following contract type 2 Learning (TT1) coinvested payments:
+	Then the payment source component will generate the following contract type 2 coinvested payments:
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - multiple periods.feature]: add back in Learning (TT1)

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - multiple periods.feature [hunk 19]
```diff
@@ -21,7 +21,7 @@ Scenario: Contract Type 2 Learning payment
...
 
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period | ULN   | TransactionType | FundingSource			| Amount |
 	| learnref1      | 10000 | p1                     | 1      | 10000 | Learning (TT1)  | CoInvestedSfa (FS2)		| 540    |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - multiple periods.feature]: can remove learnrefnumber, ukprn and uln

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature [hunk 20]
```diff
@@ -2,10 +2,10 @@
 
 Background:
 	Given the current processing period is 1
-
+	And the payments are for the current collection year
 	And a learner with LearnRefNumber learnref1 and Uln 10000 undertaking training with training provider 10000
 
-	And the SFA contribution percentage is "90%"
+	And the SFA contribution percentage is 90%"
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature]: remove trailing "

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature [hunk 21]
```diff
@@ -2,10 +2,10 @@
...
 
 	And the required payments component generates the following contract type 2 payable earnings:
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period | ULN   | TransactionType | Amount |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature]: can remove learnrefnumber, ukprn and uln

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature [hunk 22]
```diff
@@ -19,7 +19,7 @@ Scenario: Contract Type 2 Learning payment
 
 	When required payments event is received
 
-	Then the payment source component will generate the following contract type 2 Learning (TT1) coinvested payments:
+	Then the payment source component will generate the following contract type 2 coinvested payments:
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature]: add back in Learning (TT1)

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature [hunk 23]
```diff
@@ -19,7 +19,7 @@ Scenario: Contract Type 2 Learning payment
...
 
 	| LearnRefNumber | Ukprn | PriceEpisodeIdentifier | Period | ULN   | TransactionType | FundingSource			| Amount |
 	| learnref1      | 10000 | p1                     | 1      | 10000 | Learning (TT1)  | CoInvestedSfa (FS2)		| 540    |
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature]: can remove learnrefnumber, ukprn and uln

File: src/SFA.DAS.Payments.FundingSource.Domain/Services/SfaCoInvestedPaymentProcessor.cs [hunk 24]
```diff
@@ -1,5 +1,7 @@
-﻿using SFA.DAS.Payments.FundingSource.Domain.Interface;
+﻿using System;
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.Domain/Services/SfaCoInvestedPaymentProcessor.cs]: unused using

File: src/SFA.DAS.Payments.FundingSource.NonLevyFundedService/Handlers/ApprenticeshipContractType2RequiredPaymentEventHandler.cs [hunk 25]
```diff
@@ -28,7 +28,7 @@ public async Task Handle(ApprenticeshipContractType2RequiredPaymentEvent message
         {
             using (var scope = lifetimeScope.BeginLifetimeScope())
             {
-                paymentLogger.LogInfo($"Processing CalculatedPaymentDueEvent Service event. Message Id : {context.MessageId}");
+                paymentLogger.LogInfo($"Processing Required Payment Service event for Message Id : {context.MessageId}");
 
                 var executionContext = (ESFA.DC.Logging.ExecutionContext)scope.Resolve<IExecutionContext>();
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.FundingSource.NonLevyFundedService/Handlers/ApprenticeshipContractType2RequiredPaymentEventHandler.cs]: could this throw an exception if there isn't an execution context? would it be better to be defensive and do "as"?

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature (hunk 22 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature]: We already have them in the result table so I am using that.

File: src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature (hunk 23 above)
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.FundingSource.AcceptanceTests/Tests/ACT2_TransTypes_1to3_Basic_Tests/Learning (TT1) only - single period.feature]: Will update the feature file in Small Employers story

**Pull request merged**